packages/legacy/core/App/utils/helpers.ts

```diff
@@ -1017,24 +1018,36 @@ export const connectFromScanOrDeepLink = async (
         params: { threadId: receivedInvitation?.outOfBandRecord.outOfBandInvitation.threadId },
       })
     }
-    // try unsupported deeplink if built-in Credo methods fail. Let this catch block throw any error, it will be caught a level up
-  } catch (err: unknown) {
-    logger.error('Error connecting from invitation, trying unsupported query params. Error:', err as Error)
+
+    // try unsupported query params if built-in Credo methods fail. Let this catch block throw, it will be caught a level up
+  } catch (initialErr: unknown) {
+    logger.error('Error connecting from invitation, trying unsupported query params. Error:', initialErr as Error)
```

Comment:
I think this is getting a little unruley with nested try/catches. Can you extract some logic into functions can call them? The function names `performMainOperation` and `performFallbackOperation` are just examples below.

```
try {
    // This might fail
    performMainOperation();
} catch (error) {
    console.error("Main operation failed", error);
    try {
        // Fallback operation
        performFallbackOperation();
    } catch (secondaryError) {
        console.error("Fallback operation also failed", secondaryError);
        // Throw if needed
    }
}
```

Another pattern might be:

```
try {
   const success = await performMainOperation()
   if (success) {
       return
   }

    await performFallbackOperation()
catch (...) {
 
}
```

Reply:
True. I'll work on that

Reply:
@jleach cleaned it up and updated the JSDoc comments, let me know what you think